core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3427,6 +3427,33 @@ private void processJoinConditions(
         }
     }
 
+    private void propagateHints(QueryModel model) {
+        if (model == null) {
```

Comment:
nit: this is a classic traversal method, if sig was ``` propagateHintsTo(tagretModel. hints)``` we'd do a lot less null checking and make this method more self-explanatory